hnn_core/gui.py

```diff
@@ -0,0 +1,1028 @@
+"""IPywidgets GUI."""
+
+# Authors: Mainak Jas <mjas@mgh.harvard.edu>
+#          Huzi Cheng <hzcheng15@icloud.com>
+
+import codecs
+import multiprocessing
+import os.path as op
+import numpy as np
+import os
+import matplotlib.pyplot as plt
+from IPython.display import display
+from ipywidgets import (HTML, Accordion, AppLayout, BoundedFloatText, Button,
+                        Dropdown, FileUpload, FloatLogSlider, FloatText, HBox,
+                        IntText, Layout, Output, RadioButtons, Tab, Text, VBox,
+                        interactive_output, GridspecLayout)
+
+import hnn_core
+from hnn_core import (MPIBackend, JoblibBackend, read_params, simulate_dipole,
+                      jones_2009_model)
+from hnn_core.params import (_read_json, _read_legacy_params,
+                             _extract_drive_specs_from_hnn_params)
+from hnn_core.viz import plot_dipole
+
+
+def create_expanded_button(description, button_style, height):
+    style = {'button_color': '#8A2BE2'}
+    return Button(description=description,
+                  button_style=button_style,
+                  layout=Layout(height=height, width='auto'),
+                  style=style)
+
+
+def _get_sliders(params, param_keys):
+    """Get sliders"""
+    style = {'description_width': '150px'}
+    sliders = list()
+    for d in param_keys:
+        slider = FloatLogSlider(value=params[d],
+                                min=-5,
+                                max=1,
+                                step=0.2,
+                                description=d.split('gbar_')[1],
+                                disabled=False,
+                                continuous_update=False,
+                                orientation='horizontal',
+                                readout=True,
+                                readout_format='.2e',
+                                style=style)
+        sliders.append(slider)
+
+    def _update_params(variables, **updates):
+        params.update(dict(**updates))
+
+    interactive_output(_update_params, {s.description: s for s in sliders})
+    return sliders
+
+
+def _get_cell_specific_widgets(
+    layout,
+    style,
+    location,
+    data=None,
+):
```

Comment:
try to keep the style consistent with the rest of the codebase. In function definitions, try to fit as many arguments in a line as allowed by pep8 79 character limit.